packages/docusaurus-theme-classic/src/theme/StructuredData/index.tsx

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+
+import type {Props} from '@theme/StructuredData';
+
+export default function StructuredData({structuredData}: Props): JSX.Element {
+  return (
+    <script
+      type="application/ld+json"
+      // We're using dangerouslySetInnerHTML because we want to avoid React
+      // transforming quotes into &quot; which upsets parsers.
+      // The entire contents is a stringified JSON object so it is safe
```

Comment:
can you explain how to reproduce that problem here?

Was the code we documented before affected by any issue?

```tsx
        <script type="application/ld+json">
          {JSON.stringify({
            '@context': 'https://schema.org/',
            '@type': 'Organization',
            name: 'Meta Open Source',
            url: 'https://opensource.fb.com/',
            logo: 'https://opensource.fb.com/img/logos/Meta-Open-Source.svg',
          })}
        </script>
```

Can you show side-by-side examples in a repro, before/after, rendering differently in practice? And explain how it upsets parsers?

Reply:
So this was a curious one. It surfaced in the Google Search Console, and relates to the unsuccessful parsing of the inner JSON when rendered internally to the component.  This error would surface:

> Parsing error: Missing '}' or object member name.

I believe this is because by *not* using the `dangerouslySetInnerHtml` approach, the `"` characters in the JSON-LD were being rendered as `&quot;` - which is not valid JSON. So something like this:

```
        <script type="application/ld+json">
          {
            &quot;@context&quot;: &quot;https://schema.org/&quot;,
            &quot;@type&quot;: &quot;Organization&quot;,
            &quot;name&quot;: &quot;Meta Open Source&quot;,
            &quot;url&quot;: &quot;https://opensource.fb.com/&quot;,
            &quot;logo&quot;: &quot;https://opensource.fb.com/img/logos/Meta-Open-Source.svg&quot;,
          }
        </script>
```

When I migrated to the `dangerouslySetInnerHtml` approach, this fixed it in the Google Search Console:

![image](https://github.com/facebook/docusaurus/assets/1010525/232cf5dc-b0f1-4489-b830-26a88963689b)